src/Application/Products/Commands/AddProduct/AddProductCommandHandler.cs

```diff
@@ -44,7 +44,7 @@ public async Task<ProductDto> Handle(AddProductCommand request, CancellationToke
 
         try
         {
-            await _stockClient.UpdateStock(product.Id, 1);
+            await _stockClient.UpdateStock(product.Id.Value, 1);
         }
         catch (Exception ex)
         {
```

Comment:
On next line
```suggestion
        {
        _logger.LogError(ex, "Error updating the stock for the product {Id}", product.Id.Value);
```

Reply:
This is one of the "fears" of using `ValueObjects`, having to access the `Value` property.
I was thinking, is there a way to go directly if only one property inside, just referencing the `ValueObject`?

Reply:
I think not even using the library ValueOf you can get rid of this issue....
If you find any solution, it will be more than welcome :-)

Reply:
I thought about it yesterday, and nothing reasonable came up to my mind. Probably something could be done in the future, we can discuss about it :)